agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+
+# 1.b.i. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+)
+
+# 1.b.ii. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+    environment_slug="...",
+)
+
+# 2.a.i. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_reference="...",
+)
+
+# 2.a.ii. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_reference="...",
+    environment_slug="...",
+)
+
+# 2.b.i. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment="...",
+)
+
+# 2.b.ii. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment="...",
+    environment_slug="...",
+)
+
+##### FETCH #####
+
+# 1. existing environment
+environment: Environment = ag.environments.fetch(
+    environment_reference="...",
+)
+
+##### DEPLOY #####
+
+# 1.a. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant_reference="...",
+)
+
+# 1.b. existing variant
+environment: Environment = ag.prompts.deploy(
+    variant="...",
+)
+
+# 2.a. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_slug="...",
+    environment_reference="...",
+)
+
+# 2.b. existing environment
+environment: Environment = ag.prompts.deploy(
+    environment_slug="...",
+    environment="...",
+)
+
+
+########## PROMPTS ##########
+
+##### CREATE #####
+
+# 1.a.i. new prompt
+prompt: Prompt = ag.prompts.create(
+    parameters="...",
+)
+
+# 1.a.ii. new prompt
+prompt: Prompt = ag.prompts.create(
+    parameters="...",
+    slug="...",
+)
+
+# 1.a.iii. new prompt
+prompt: Prompt = ag.prompts.create(
+    parameters="...",
+    slug="...",
+    application_slug="...",
+)
+
+# 1.a.iv. new prompt
+prompt: Prompt = ag.prompts.create(
+    parameters="...",
+    slug="...",
+    application_slug="...",
+    template_key="...",
+)
+
+# 2.a.i. existing prompt
+prompt: Prompt = ag.prompts.create(
+    prompt_reference="...",
+)
+
+# 2.a.ii. existing prompt
+prompt: Prompt = ag.prompts.create(
+    prompt_reference="...",
+    slug="...",
+)
+
+# 2.b.i. existing prompt
+prompt: Prompt = ag.prompts.create(
+    prompt="...",
+)
+
+# 2.b.ii. existing prompt
+prompt: Prompt = ag.prompts.create(
+    prompt="...",
+    slug="...",
+)
+
+##### FETCH #####
+
+# 1. existing prompt
+prompt: Prompt = ag.prompts.fetch(
+    prompt_reference="...",
+)
+
+# 2. existing environment
+prompt: Prompt = ag.prompts.fetch(
+    environment_slug="...",
+)
+
+##### DEPLOY #####
+
+# 1.a. existing environment
+environment: Environment = ag.prompts.deploy(
+    prompt_reference="...",
+    environment_slug="...",
+)
+
+# 1.b. existing environment
+environment: Environment = ag.prompts.deploy(
+    prompt="...",
+    environment_slug="...",
+)
+
+# 2.a. new environment
+environment: Environment = ag.prompts.deploy(
+    prompt_reference="...",
+)
+
+# 2.b. new environment
+environment: Environment = ag.prompts.deploy(
```

Comment:
why does prompts only can be deployed? (or is it just an artefact from chatgpt)

Reply:
Because, _Prompts_ is the only one that I meant to design as wrapper, with methods that do plenty of things in the background. Others like _Variants_, and only in this baseline SDK, follow more of a _CRUD_ type of interface where deploying a _Variant_ happens through an _Environment_.